lib/matplotlib/tests/test_ticker.py

```diff
@@ -671,6 +778,181 @@ def test_LogFormatter_call(self, val):
         assert temp_lf(val) == str(val)
 
 
+class TestLogitFormatter:
+    @staticmethod
+    def logit_deformatter(string):
+        r"""
+        Parser to convert string as r'$\mathdefault{1.41\cdot10^{-4}}$' in
+        float 1.41e-4, as '0.5' or as r'$\mathdefault{\frac{1}{2}}$' in float
+        0.5,
+        """
+        match = re.match(
+            r"[^\d]*"
+            r"(?P<comp>1-)?"
+            r"(?P<mant>\d*\.?\d*)?"
+            r"(?:\\cdot)?"
+            r"(?:10\^\{(?P<expo>-?\d*)})?"
+            r"[^\d]*$",
+            string,
+        )
+        if match:
+            comp = match["comp"] is not None
+            mantissa = (
+                float(match["mant"])
+                if (match["mant"] is not None and len(match["mant"]) > 0)
```

Comment:
that's just `if match["mant"]`?  same below.

Reply:
Yes. It was incremental work, I start with `match[…]` is not None` and I add constraints…
Changed.